.cursor/rules/unit-test-rules.mdc

```diff
@@ -24,3 +24,30 @@ alwaysApply: false
 - Test custom hooks with `renderHook` when isolated testing is needed
 - Use descriptive assertion messages when the test failure might be unclear
 - Avoid try/catch blocks in unit tests
+
+## CRITICAL: Always Use Existing Test Utilities Before Creating New Mocks
+
+**Before creating new mocks, check `~/testUtils` for existing utilities:**
+
+- **Feature Gates**: ALWAYS use `mockUseFeatureGate([[FEATURE_NAME, true/false]])` from `~/testUtils`. NEVER manually mock `useFetchFeatureGate` or `useFeatureGate`.
+  - Example: `mockUseFeatureGate([[MULTIREGION_PREVIEW_ENABLED, true]])`
+  - Import: `import { mockUseFeatureGate } from '~/testUtils'`
+
+- **Chrome/Red Hat Services**: Use `mockUseChrome(mockImpl?)` from `~/testUtils` instead of manually mocking `@redhat-cloud-services/frontend-components/useChrome`
+
+- **Restricted Environment**: Use `mockRestrictedEnv(mockValue)` from `~/testUtils` instead of manually mocking restricted environment utilities
+
+- **Redux State**: Use `withState(initialState)` from `~/testUtils` for Redux testing instead of manually creating stores
+
+- **Formik/useFormState**: When mocking `useFormState` from `~/components/clusters/wizards/hooks`:
+  1. Mock at module level: `jest.mock('~/components/clusters/wizards/hooks')`
+  2. Import the hook: `import { useFormState } from '~/components/clusters/wizards/hooks'`
+  3. Create typed mock at top of test suite: `const mockedUseFormState = useFormState as jest.MockedFunction<typeof useFormState>`
+  4. Use `mockedUseFormState.mockReturnValue({ ... })` per test, NOT `(useFormStateModule.useFormState as jest.Mock)`
+  5. See `src/components/clusters/common/Upgrades/UpgradeSettingsFieds.test.tsx` for reference pattern
```

Comment:
This is not a good idea. 

I can understand mocking external systems we depend on (`useChrome`, feature gates) or a global state like redux (until it's gone). But formik is a local tool and we shouldn't need to mock it. When needed we can just wrap components with a dummy formik form inside the test. We shouldn't test our components implementation details.

The example mentioned here doesn't need any formik mocking at all.